interop/tapir/src/main/scala/caliban/interop/tapir/HttpAdapter.scala

```diff
@@ -0,0 +1,135 @@
+package caliban.interop.tapir
+
+import caliban._
+import caliban.interop.tapir.TapirAdapter._
+import sttp.model.{ headers => _, _ }
+import sttp.tapir.Codec.JsonCodec
+import sttp.tapir.model.ServerRequest
+import sttp.tapir.server.ServerEndpoint
+import sttp.tapir.{ headers, _ }
+import zio._
+
+sealed trait HttpAdapter[-R, E] { self =>
+  protected val endpoints: List[PublicEndpoint[(GraphQLRequest, ServerRequest), TapirResponse, GraphQLResponse[E], Any]]
+
+  protected def executeRequest(
+    graphQLRequest: GraphQLRequest,
+    serverRequest: ServerRequest
+  ): ZIO[R, TapirResponse, GraphQLResponse[E]]
+
+  def serverEndpoints[R1 <: R]: List[ServerEndpoint[Any, RIO[R1, *]]] = {
+    def logic(request: (GraphQLRequest, ServerRequest)): RIO[R1, Either[TapirResponse, GraphQLResponse[E]]] = {
+      val (graphQLRequest, serverRequest) = request
+      executeRequest(graphQLRequest, serverRequest).either
+    }
+    endpoints.map(_.serverLogic(logic))
+  }
+
+  def configure[R1](configurator: ZLayer[R1 & ServerRequest, TapirResponse, R]): HttpAdapter[R1, E] =
+    HttpAdapter.Configured(self, configurator)
+
+  def configure(configurator: URIO[Scope, Unit]): HttpAdapter[R, E] =
```

Comment:
Just for my own understanding, how is this meant to be used?

Reply:
`adapter.configure(Configurator.setSkipValidationScoped(true))`

Using a `ZIO` is a little easier to write than a `ZLayer` when you don't use the incoming part of the environment.